### PR TITLE
Java Servlets

### PR DESCRIPTION
**Pull request by linamontes10-zz**
Add asynchronous JS function to fetch data from server and add it to a different page. Update to reflect the changes made in the main STEP 2020 repository.

### CODE REVIEW

File: portfolio/src/main/webapp/auth.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+BQCXwOL0Kb5H0nAc1iOaM4AQ4obcCYVHi8C6IgHZr8Z8oHEE-fDzqhkNEk_yxUXa6JOsxIYJoXfdNrLJ-61OtVEYOAXAUhQC83gqSsQMJ8zzq1a3Z0FbV0XX-bJdJZISe_DwHliFohhBXnKnUSAHXtsy8q6Yg2IvKiFA3rGl5vcaMw
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/auth.txt]: I'm not sure exactly what this is but my guess is you should .gitignore it

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -28,11 +28,17 @@ function addRandomFunFact() {
...
     const funFact = funFacts[Math.floor(Math.random() * funFacts.length)];
 
     // Add it to the page.
     const funFactContainer = document.getElementById('fun-fact-container');
     funFactContainer.innerText = funFact;
 }
+
+async function addContent() {
+  const response = await fetch('/data');
+  const hello = await response.text();
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/script.js]: nit: Give this variable a more descriptive name

File: walkthroughs/week-3-server/examples/favorite-color/pom.xml [hunk 3]
```diff
@@ -28,9 +28,13 @@
   <build>
     <plugins>
       <plugin>
-        <groupId>com.google.appengine</groupId>
+        <groupId>com.google.cloud.tools</groupId>
         <artifactId>appengine-maven-plugin</artifactId>
-        <version>1.9.71</version>
+        <version>2.2.0</version>
+        <configuration>
+          <deploy.projectId>YOUR_PROJECT_ID_HERE</deploy.projectId>
```
Comment (KeatonTech) [inline walkthroughs/week-3-server/examples/favorite-color/pom.xml:35]: Are these changes to the example projects intentional? They still don't have the project id specified so I can't imagine this actually works

**Review by KeatonTech — CHANGES_REQUESTED** [review]

File: walkthroughs/week-3-server/examples/favorite-color/pom.xml (hunk 3 above)
Comment (linamontes10-zz) [reply walkthroughs/week-3-server/examples/favorite-color/pom.xml:35]: These changes are from the main STEP repository, so I think we as the interns add the project id to the file as we go through the trainings.

Comment (KeatonTech) [reply walkthroughs/week-3-server/examples/favorite-color/pom.xml:35]: Ah, gotcha. So you intentionally pulled these in from the STEP repo? That makes sense, in the future I'd prefer that be a separate PR though, one that doesn't also include your own changes.

**Review by KeatonTech — APPROVED** [review]

File: walkthroughs/week-3-server/examples/favorite-color/pom.xml (hunk 3 above)
Comment (linamontes10-zz) [reply walkthroughs/week-3-server/examples/favorite-color/pom.xml:35]: I will definitely make sure you pull in any future changes in a separate PR next time, sorry about that!

**Pull request merged**